package/src/components/AttachmentPicker/AttachmentPicker.tsx

```diff
@@ -237,9 +237,13 @@ export const AttachmentPicker = React.forwardRef(
       // `id` is available for Expo MediaLibrary while Cameraroll doesn't share id therefore we use `uri`
       selected:
         selectedImages.some((image) =>
-          image.id ? image.id === asset.id : image.uri === asset.uri,
+          image.id
+            ? image.id === asset.id
+            : image.uri === asset.uri || image.originalUri === asset.uri,
```

Comment:
Why is the originalUri condition needed here?

Reply:
We typically want to check for both if people want to override the way the `uri` is being generated. I was also considering removing it but it's better that it's there for safety. It can't hurt us anyway.